pkg/cloudprovider/providers/azure/azure.go

```diff
@@ -530,3 +569,76 @@ func (az *Cloud) GetActiveZones() (sets.String, error) {
 func (az *Cloud) GetLocation() string {
 	return az.Location
 }
+
+// GetNodeResourceGroup gets resource group for given node.
+func (az *Cloud) GetNodeResourceGroup(nodeName string) (string, error) {
+	// Kubelet won't set az.nodeInformerSynced, always return configured resourceGroup.
+	if az.nodeInformerSynced == nil {
+		return az.ResourceGroup, nil
+	}
+
+	az.nodeCachesLock.Lock()
```

Comment:
it's a get operation, is it necessary to put a lock here? I am worried about too many lock here